add helm-rg recipe

cosmicexplorer opened the pull request:
### Brief summary of what the package does

[`helm-rg`](https://github.com/cosmicexplorer/helm-rg) exposes a [`helm`](https://github.com/emacs-helm/helm) interface to [`ripgrep`](https://github.com/BurntSushi/ripgrep). `ripgrep` is invoked as an asynchronous process, updating the search results as you type.

### Direct link to the package repository

https://github.com/cosmicexplorer/helm-rg

### Your association with the package

I am the maintainer of this package.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). 
- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

purcell commented:
Thanks! Quick feedback:
- The recipe will include `update-commentary.el`. If that's a tool, put it in a subdirectory of your repo.
- Since this is helm-specific, why not call it "helm-rg"?
- The `M-d` and `M-g` key bindings are a bit random, and since they have common meanings, I'd expect they might be confusing. (Disclaimer: I'm not familiar with helm conventions, though.)
- The helm source is saved in a private var, but isn't it possible to expose it for general use, ie. not necessarily via the `rg3` wrapper command?
- `cl-mapc` can just be `mapc`, which is a standard built-in.

cosmicexplorer commented:
Wow, thanks a lot (seriously) for the very helpful comments!

> The recipe will include update-commentary.el. If that's a tool, put it in a subdirectory of your repo.

Whoops! Moved to a new [`gen-commentary/` subdirectory](https://github.com/cosmicexplorer/helm-rg/tree/master/gen-commentary).

> Since this is helm-specific, why not call it "helm-rg"?

Thanks a lot -- I hadn't considered that and it makes a lot of sense. I have no allegiance to the previous name, so I've renamed it to `helm-rg` (also reflected in the code and the PR title/post content).

> The M-d and M-g key bindings are a bit random, and since they have common meanings, I'd expect they might be confusing. (Disclaimer: I'm not familiar with helm conventions, though.)

Neither am I! This seems perfectly reasonable -- to clarify, are there obvious, specific keybindings you have in mind for either changing the current directory or the file glob pattern? If so, I would want to use those, otherwise I'll search around a bit to get a better idea of how to map these keys.

> The helm source is saved in a private var, but isn't it possible to expose it for general use, ie. not necessarily via the rg3 wrapper command?

Absolutely -- I hadn't considered that use case at all. I've changed the name of `rg3--process-source` into `helm-rg-process-source`. Is that what you meant?

> cl-mapc can just be mapc, which is a standard built-in.

Didn't know that! Fixed.

purcell commented:
> Absolutely -- I hadn't considered that use case at all. I've changed the name of rg3--process-source into helm-rg-process-source. Is that what you meant?

Yep.

> are there obvious, specific keybindings you have in mind for either changing the current directory or the file glob pattern? If so, I would want to use those, otherwise I'll search around a bit to get a better idea of how to map these keys.

Not really. Take a look around and see if there's a similar package you could crib from. :-)

In the meantime, I'll go ahead and merge this.
